bliss/models/galaxy_net.py

```diff
@@ -65,30 +67,42 @@ def __init__(
         super().__init__()
         self.save_hyperparameters()
 
-        self.main_encoder = CenteredGalaxyEncoder(slen=slen, latent_dim=latent_dim, n_bands=n_bands)
-        self.main_decoder = CenteredGalaxyDecoder(slen=slen, latent_dim=latent_dim, n_bands=n_bands)
+        self.main_encoder = CenteredGalaxyEncoder(
+            slen=slen,
+            latent_dim=(latent_dim // 2) * 2,
+            n_bands=n_bands,
+        )
+        self.main_decoder = CenteredGalaxyDecoder(
+            slen=slen, latent_dim=latent_dim // 2, n_bands=n_bands
+        )
         self.main_autoencoder = nn.Sequential(self.main_encoder, self.main_decoder)
 
         self.residual_encoder = CenteredGalaxyEncoder(
-            slen=slen, latent_dim=latent_dim, n_bands=n_bands
+            slen=slen,
+            latent_dim=(latent_dim // 2) * 2,
+            n_bands=n_bands,
         )
         self.residual_decoder = CenteredGalaxyDecoder(
-            slen=slen, latent_dim=latent_dim, n_bands=n_bands
+            slen=slen, latent_dim=latent_dim // 2, n_bands=n_bands
         )
         self.residual_autoencoder = nn.Sequential(self.residual_encoder, self.residual_decoder)
 
+        self.register_buffer("prior_mean", torch.tensor(0.0))
+        self.register_buffer("prior_var", torch.tensor(1.0))
+        self.dist_main = Normal(self.prior_mean, self.prior_var)
+        self.dist_residual = Normal(self.prior_mean, self.prior_var)
+
         self.residual_delay_n_steps = residual_delay_n_steps
         assert slen == 53, "Currently slen is fixed at 53"
         self.slen = slen
-        assert latent_dim == 64, "Currently latent_dim is fixed at 64"
         self.latent_dim = latent_dim
         self.min_sd = min_sd
 
     def forward(self, image, background):
         """Gets reconstructed image from running through encoder and decoder."""
-        recon_mean_main = self._main_forward(image, background)
-        recon_mean_residual = self._residual_forward(image - recon_mean_main)
-        return recon_mean_main + recon_mean_residual
+        recon_mean_main, pq_latent_main = self._main_forward(image, background)
+        recon_mean_residual, pq_latent_residual = self._residual_forward(image - recon_mean_main)
+        return recon_mean_main + recon_mean_residual, pq_latent_main + pq_latent_residual
```

Comment:
If this is now a VAE, should it also return the variance?

Reply:
It returns a single sample and the difference between the prior and variational density for that sample (pq_latent_*). I kept it this way to require less changes from what we had before. I will likely refactor this when we want to draw multiple samples at once (#363).